### PR TITLE
Windows workflow

### PR DESCRIPTION
**Pull request by SiziweLupondo**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Additional Context (Please include any Screenshots/gifs if relevant)

...

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have tagged the reviewers in a comment below incase my pull request is ready for a review
- [x] I have signed the commit message to agree to Developer Certificate of Origin (DCO) (to certify that you wrote or otherwise have the right to submit your contribution to the project.) by adding "--signoff" to my git commit command.

### CODE REVIEW

File: .github/workflow/windows.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+name: Build and Package FLINT for Windows
+
+on:
+  push:
+    branches:
+      - develop
```
Comment (Simpleshell3) [inline .github/workflow/windows.yml]: for now, also add the Windows_workflow branch. This way you can run the workflow and see if there are any errors before merging.

**Review by Simpleshell3 — COMMENTED** [review]
See comment

File: .github/workflow/windows.yml (hunk 1 above)
Comment (Simpleshell3) [reply .github/workflow/windows.yml]: also directory name: workflow ---> workflows

**Comment by SiziweLupondo** [comment]
@Simpleshell3 I have made the changes suggested, Please advise.

**Comment by Simpleshell3** [comment]
it is beginning to look good now. You can see the workflow runs from here: https://github.com/SiziweLupondo/FLINT/actions and check why they are failing.

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    name: Build and Package FLINT for Windows
+    runs-on: windows-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: Windows_workflow  # Checkout the Windows_workflow branch
+
+      - name: Install CMake
```
Comment (Simpleshell3) [inline .github/workflows/windows.yml]: the windows runner comes with cmake, visual studio and most of the build tools.

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          Invoke-WebRequest -Uri https://github.com/git-for-windows/git/releases/download/v2.41.0.windows.1/$gitInstaller -OutFile $gitInstaller
+          Start-Process $gitInstaller -ArgumentList "/VERYSILENT", "/NORESTART" -NoNewWindow -Wait
+
+      - name: Install Visual Studio Build Tools
+        run: |
+          $vsInstaller = "vs_buildtools.exe"
+          Invoke-WebRequest -Uri https://aka.ms/vs/16/release/vs_buildtools.exe -OutFile $vsInstaller
+          Start-Process $vsInstaller -ArgumentList "--quiet", "--wait", "--norestart", "--nocache", "--installPath", "C:\BuildTools", "--add", "Microsoft.VisualStudio.Workload.VCTools;includeRecommended" -NoNewWindow -Wait
+
+      - name: Install Boost libraries
```
Comment (Simpleshell3) [inline .github/workflows/windows.yml]: hmm i also noticed @mfellows has this updated script that installs dependencies and builds flint. I think we can call the script in the workflow

**Review by Simpleshell3 — COMMENTED** [review]
Everything looks good, but could be better. Try using the following workflow. It is a bit rough but will give you a good starting point.
```

name: Build and Package FLINT for Windows with VCPKG

on:
  push:
    branches: 
          - main
          - master
          - develop
          - Windows_workflow
  pull_request:
    branches: 
           - main
           - master
           - develop
           - Windows_workflow

jobs:
  build:
    runs-on: windows-latest

    steps:
    - name: Checkout repository
      uses: actions/checkout@v2

    - name: Set up Visual Studio 2022
      uses: ilammy/msvc-dev-cmd@v1
      with:
        arch: x64
        version: 17  

    - name: Cache vcpkg
      uses: actions/cache@v3
      with:
        path: vcpkg
        key: vcpkg-${{ runner.os }}-${{ hashFiles('Source/vcpkg.json') }}
        restore-keys: |
          vcpkg-${{ runner.os }}-

    - name: Install and run Dependencies
      shell: pwsh
      run: |
         pushd vcpkg
         .\vcpkg_setup_vs2022_win_x64.bat
         popd
         
    - name: Create executable package
      shell: pwsh
      run: |
        pushd Source/build
        ls
        Compress-Archive -Path * -DestinationPath moja-flint.zip -Force 
        Move-Item -Path *.zip -Destination ../../moja-flint.zip 
        popd

    - name: Upload build artifacts
      uses: actions/upload-artifact@v2
      with:
        name: moja-flint
        path: moja-flint.zip

    - name: Clean up after build
      shell: pwsh
      run: |
        Remove-Item -Recurse -Force Source\build
```

**Pull request closed without merging**